pkg/xds/generator/admin_proxy_generator_test.go

```diff
@@ -93,16 +96,29 @@ var _ = Describe("AdminProxyGenerator", func() {
 			expected:      "03.envoy-config.golden.yaml",
 			adminAddress:  "::1",
 		}),
-		Entry("should generate admin resources, unspecified IPv4", testCase{
+		Entry("should generate admin resources, unspecified IPv4, readiness port 0", testCase{
 			dataplaneFile: "04.dataplane.input.yaml",
 			expected:      "04.envoy-config.golden.yaml",
 			adminAddress:  "0.0.0.0",
+			readinessPort: 0,
 		}),
 		Entry("should generate admin resources, unspecified IPv6", testCase{
 			dataplaneFile: "05.dataplane.input.yaml",
 			expected:      "05.envoy-config.golden.yaml",
 			adminAddress:  "::",
 		}),
+		Entry("should generate admin resources, IPv4 with readiness port 9902", testCase{
+			dataplaneFile: "04.dataplane.input.yaml",
+			expected:      "06.envoy-config.golden.yaml",
+			adminAddress:  "127.0.0.1",
+			readinessPort: 9902,
+		}),
+		Entry("should generate admin resources, IPv6 with readiness port 9400", testCase{
+			dataplaneFile: "05.dataplane.input.yaml",
+			expected:      "07.envoy-config.golden.yaml",
+			adminAddress:  "::1",
+			readinessPort: 9400,
```

Comment:
Is this enabled by default? The above examples don't have a defined value, and it seems to use the old way or is it just a test?

Reply:
This is a test case and I'm adding a value to assert this value should be used.

Reply:
btw, it's enabled by default from the DP. But from CP's perspective, it does not have a default value and relies on the port value passed form the DP to decide whether it should generate xds config to forward to the readiness reporter component.